test/task.cc

```diff
@@ -76,6 +77,37 @@ TEST(Task, Void) {
   EXPECT_EQ(100, x);
 }
 
+TEST(Task, CatchesFinally) {
+  auto f = []() -> Task::Of<int>::Catches<std::runtime_error> {
+    return []() {
+      return Finally([](expected<
+                         void,
+                         std::variant<
+                             Stopped,
+                             std::runtime_error>>&& expected) {
+               return Just(100);
+             })
+          >> Just(10);
```

Comment:
How about we remove this so that we are confident that we execute the `Finally`?
```suggestion
             });
```